OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -162,6 +162,9 @@ public class BaseBuilderBotModuleInfo : ConditionalTraitInfo, NotBefore<Resource
 		[Desc($"AI will move the only mcv when those numbers of refinery <= productions + tech - {nameof(ForceExpansionTolerate)}.")]
 		public readonly ImmutableArray<int> ForceExpansionTolerate = [2, 3];
 
+		[Desc("Increase the expansion tolerate by 1 for this many cash.")]
```

Comment:
```suggestion
		[Desc("Decrease the expansion tolerate by Cash / this. Used to prevent AI from expanding when it has enough cash.")]
```

Reply:
> It seems no there's no mechanism that makes AI expand when overflowing with cash? Expanding isn't just for cash.

There is actually, 
1. see "McvExpansionManagerBotModule"'s "MoveConyardTick". this is the default expansion cooldown when bot has more than 2 mcvs.

2. see "BaseBuilderQueueManager"'s line 65, when bot find there is not enough place to place building, it will try unpack and find a new place (you can have a test)

Those are probably suffcient enough for bot on expansion that unrelated with economy.